### DISCUSSION TITLE
Is there way to inject user_id in create_schema using dependency injection based on user passed on create_route dependency??

### DISCUSSION THREAD

**Discussion by bhuwanaryal19** [awtkns/fastapi-crudrouter#183] [Q&A]
Here I want to inject owner_id based on current user session. 
```
from pydantic import BaseModel
from fastapi import FastAPI
from fastapi_crudrouter import MemoryCRUDRouter as CRUDRouter
from app.oauth2 import get_current_user
import fastapi_users

current_user = fastapi_users.current_user()

class Potato(BaseModel):
    id: int
    color: str
    mass: float
    owner_id: Optional[UUID]  # inject automatically using Dependency Injection


# class CreatePotato(BaseModel):


app = FastAPI()
app.include_router(CRUDRouter(schema=Potato), create_route = [Depends(current_user)])
```

**Comment by hasansezertasan** [comment]
Let's say that I have two tables in my database.

- Representing Data for the `users` table:

| id  | username | password |
| --- | -------- | -------- |
| 1   | user1    | pass1    |
| 2   | user2    | pass2    |
| 3   | user3    | pass3    |

- Representing Data for the `posts` table:

| id  | title | content | user_id |
| --- | ----- | ------- | ------- |
| 1   | post1 | content | 1       |
| 2   | post2 | content | 1       |
| 3   | post3 | content | 2       |
| 4   | post4 | content | 3       |
| 5   | post5 | content | 3       |

And I'm using JWT for authentication, which helps me find out who the user is.

What I want to do is:
  - FastAPI CRUD Router provides a route that returns all the posts in my database and I can paginate with it, awesome. But I also want a route that will return all posts of the user who is logged in. I can't do that with FastAPI CRUD Router. The great library gives me the ability to check if the user logged in with the dependency injection system but it can't separate the users. My goal here is to return the data by the `user_id` of the user which can be taken from the JWT token.
  - If a user tries to delete or update a post he/she doesn't own, I want to return a 403 Forbidden error.

Let me know if you have a solution for this.

**Reply by derobiom** [reply]
I am trying to do something similar. I want to plug in the user_id of the current user as a foreign key in another model. I can get it in the main.py. But how do I get it from there into the database? I don't want the client to be able to tell me which user_id they are because that could result in "cross posting". Everything I have tried seems to result in a circular reference.

**Reply by hasansezertasan** [reply]
I believe you should customize [`fastapi_crudrouter/core/sqlalchemy.py`](https://github.com/awtkns/fastapi-crudrouter/blob/9b829865d85113a3f16f94c029502a9a584d47bb/fastapi_crudrouter/core/sqlalchemy.py).

I stopped using fastapi-crudrouter because of this issue, hope they'll solve this one day.
